File: playground/ios/NucleusApp/NucleusApp/DemoModels.swift
import SwiftUI

struct TokenSwatch: Identifiable {
    let id: String
    let name: String
    let hex: String
}

struct TokenGroup: Identifiable {
    let id: String
    let name: String
    let tokens: [TokenSwatch]
}

struct SemanticTheme {
    let background: String
    let surface: String
    let surfaceAlt: String
    let border: String
    let text: String
    let muted: String
    let accent: String
    let accentContent: String
}

struct SemanticMode: Identifiable {
    let id: String
    let name: String
    let theme: SemanticTheme
    let groups: [TokenGroup]
}

struct DemoSection: Identifiable {
    let id: String
    let label: String
    let groups: [TokenGroup]
}

enum DemoAppearance: String, CaseIterable, Identifiable {
    case light
    case dark

    var id: String { rawValue }

    var name: String {
        switch self {
        case .light:
            return "Light"
        case .dark:
            return "Dark"
        }
    }

    var colorScheme: ColorScheme {
        switch self {
        case .light:
            return .light
        case .dark:
            return .dark
        }
    }

    var theme: SemanticTheme {
        semanticModes.first(where: { $0.id == rawValue })?.theme ?? neutralPalette
    }
}

File: examples/android/app/src/main/kotlin/com/worldcoin/nucleusapp/DemoModels.kt
package com.worldcoin.nucleusapp

import androidx.compose.ui.graphics.Color

data class TokenTheme(
    val background: Color,
    val surface: Color,
    val surfaceAlt: Color,
    val border: Color,
    val text: Color,
    val muted: Color,
    val accent: Color,
    val accentContent: Color,
)

data class AppearanceTheme(
    val id: String,
    val name: String,
    val theme: TokenTheme,
)

data class TokenGroup(
    val id: String,
    val name: String,
    val tokens: List<TokenSwatch>,
)

data class TokenSwatch(
    val id: String,
    val name: String,
    val hex: String,
    val color: Color,
)

data class SemanticModeData(
    val id: String,
    val name: String,
    val theme: TokenTheme,
    val groups: List<TokenGroup>,
)

data class TokenSectionData(
    val id: String,
    val label: String,
    val groups: List<TokenGroup>,
)

data class TokenCatalog(
    val appearanceThemes: List<AppearanceTheme>,
    val tokenSections: List<TokenSectionData>,
)
